x-pack/plugins/alerting/server/routes/rule/apis/unmute_alert/transforms/transform_request_params_to_application/v1.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UnmuteAlertParams } from '../../../../../../application/rule/methods/unmute_alert/types';
+import { RewriteRequestCase } from '../../../../../lib';
+
+export const transformRequestParamsToApplication: RewriteRequestCase<UnmuteAlertParams> = ({
```

Comment:
Should this be `<UnmuteAlertParamsV1>` ?

Reply:
Ah good point. I followed the same structure of the `mute_alert` endpoint, where that type is not versioned. It seems like none of the types in [alerting/server/application/rule/methods](https://github.com/elastic/kibana/blob/2239c6e9e992efe4ccd17e7b769342f7b22d642c/x-pack/plugins/alerting/server/application/rule/methods) is versioned (guess it makes sense since they're used as parameter types of the rules client methods, which have no multiple coexisting versions? 🤔)

Reply:
The rule methods are not versioned intentionally. The transformation functions are versioned. The reason is that they transform a versioned request to a non-versioned method (application) schema. So we transform from `UnmuteAlertRequestParamsV1` (from `routes/rule/apis/unmute_alert/index.ts`) -> `UnmuteAlertParams` (from `application/rule/methods/unmute_alert/types`). I would suggest to be more explicit for the types of the params like

```

export const transformRequestParamsToApplication: RewriteRequestCase<UnmuteAlertParams> = ({rule_id: alertId,
  alert_id: alertInstanceId,
}: UnmuteAlertRequestParamsV1) => ({
  alertId,
  alertInstanceId,
});
```

Wdyt?